src/Core/Checkout/Cart/Error/Error.php

```diff
@@ -34,6 +34,11 @@ abstract public function getLevel(): int;
 
     abstract public function blockOrder(): bool;
 
+    public function blockResubmit(): bool
+    {
+        return true;
```

Comment:
I think, by default this should be
```php
return $this->blockOrder();
```
Otherwise, all notices like "your payment method has been changed" will block submission.

Reply:
No, it is `return true;` because `blockResubmit()` is optional with the defautl value of `true`, if a Custom Error class extending the Cart Error wants to **not** block the resubmit, it has to create their own `blockResubmit` method returning `false`.

Turning this to `return $this->blockOrder();` would require every single custom cart error to implement this feature, making it a breaking change, especially if customers have custom errors in their Plugins.

Reply:
Please read the attached changelog above, it is described there aswell.

Reply:
No, `blockOrder()` is an abstract method, therefore it has already been implemented by the custom error a plugin provides. If `blockResubmit()` is not implemented, then it should behave exactly the same as the `blockOrder()` did before. If the error did not block an order, it should be submissable, and if the order was blocked, then submission should not be possible.

If a plugin should want to do it, like you price change example in #2898, it can obviously be implemented, even if we do a `return $this->blockOrder()`. Only then we want to have a difference between `blockOrder()` and `blockResubmit()`. 

Yours would be the breaking change. All errors that do not override `blockResubmit()` (which are all in the beginning) would then disable the confirm button.

Reply:
Okay, I apologize, I read `return $this->blockResubmit();` 🙈 
You are totally right, fixing it.

Reply:
Fixed in 148cec8